ERCS/erc-6538.md

```diff
@@ -27,65 +27,61 @@ This contract defines an `ERC5564Registry` that stores the stealth meta-address
 The contract is specified below. A one byte integer is used to identify the stealth address scheme. This integer is used to differentiate between different stealth address schemes. A mapping from the scheme ID to it's specification is maintained at [this](../assets/eip-5564/scheme_ids.md) location.
 
 ```solidity
-pragma solidity ^0.8.17;
+pragma solidity ^0.8.20;
 
 /// @notice Registry to map an address or other identifier to its stealth meta-address.
 contract ERC5564Registry {
   /// @dev Emitted when a registrant updates their stealth meta-address.
   event StealthMetaAddressSet(
-    bytes indexed registrant, uint256 indexed scheme, bytes stealthMetaAddress
+    address indexed registrant, uint256 indexed scheme, bytes stealthMetaAddress
   );
 
   /// @notice Maps a registrant's identifier to the scheme to the stealth meta-address.
   /// @dev Registrant may be a 160 bit address or other recipient identifier, such as an ENS name.
   /// @dev Scheme is an integer identifier for the stealth address scheme.
   /// @dev MUST return zero if a registrant has not registered keys for the given inputs.
-  mapping(bytes => mapping(uint256 => bytes)) public stealthMetaAddressOf;
+  mapping(address => mapping(uint256 => bytes)) public stealthMetaAddressOf;
+  mapping(address user => uint256 nonce) public nonceOf;
+  bytes32 private constant TYPE_HASH =
+    keccak256("EIP712Domain(string name,string version,uint256 chainId,address registryContract)");
```

Comment:
These should have natspec comments, feel free to tweak my suggestions. I also made the typehash public because it may be useful for users.
```suggestion

  /// @notice Next nonce expected from `user` to use when signing for `registerKeysOnBehalf`.
  mapping(address user => uint256 nonce) public nonceOf;
  
  /// @dev EIP-712 Type hash used in `registerKeysOnBehalf`
  bytes32 public constant TYPE_HASH =
    keccak256("EIP712Domain(string name,string version,uint256 chainId,address registryContract)");
```

Reply:
Updated in the same commit as above:  https://github.com/garyghayrat/ERCs/pull/2/commits/0163a5a14a3b9517802953c5b6c7a5b68903cd4a